### PR TITLE
[PT] update specifics/plug translation

### PR DESCRIPTION
**Pull request by thiamsantos**

### CODE REVIEW

File: pt/lessons/specifics/plug.md [hunk 1]
```diff
@@ -1,35 +1,41 @@
...
 
-Se você estiver familiarizado com Ruby, você pode pensar sobre Plug como o Rack com uma pitada de Sinatra. Ele fornece uma especificação para componentes de aplicação web e adaptadores para servidores web. Mesmo não fazendo parte do núcleo de Elixir, Plug é um projeto oficial de Elixir.
+Se você estiver familiarizado com Ruby, você pode pensar sobre Plug como o Rack com uma pitada de Sinatra.
+Ele fornece uma especificação para componentes de aplicação web e adaptadores para servidores web. 
+Mesmo não fazendo parte do núcleo de Elixir, Plug é um projeto oficial de Elixir.
 
-Nós iremos começar criando uma mini aplicação web baseada no Plug. Depois, iremos aprender sobre as rotas do Plug e como adicionar um Plug à uma aplicação web existente.
+Nessa lição nós vamos construir um simples servidor HTTP do zero usando a biblioteca em Elixir `PlugCowboy`. Cowboy é um simples servidor HTTP para o Erlang e Plug vai nos disponibilizar um "connection adapter" para esse servidor web.
+
+Depos de montar nossa mini aplicação web, nós vamos apŕender as rotas do Plug e como usar vários plugs em uma única aplicação web.
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: The word `aprender` has a misspelled letter.

File: pt/lessons/specifics/plug.md [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
 $ cd example
 ```
 
+Nós precisamos que nossa aplicação Elixir inclua uma árvore de supervisão porque nós vamos usar um Supervisor para iniciar e rodar nosso servidor Cowboy2.
+
 ## Dependências
 
-Adicionar dependências é uma facilidade com mix. Para instalar Plug nós precisamos fazer duas pequenas alterações no nosso `mix.exs`.
-A primeira coisa a fazer é adicionar tanto Plug quanto um servidor web(vamos utilizar o Cowboy) no nosso arquivo de dependências:
+Adicionar dependência é muito fácil com o mix. Para usar o Plug como uma "adapter interface" para o servidor web Cowbooy2, nós precisamos instalar o pacote `PlugCowboy`:
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: Here there is a typo in `Cowboy2`.

File: pt/lessons/specifics/plug.md [hunk 3]
```diff
@@ -72,40 +78,49 @@ Ela recebe um `%Plug.Conn{}` struct como seu primeiro argumento, e é esperado q
 
 ## Configurando o módulo do projeto
 
-Dado que estamos criando uma aplicação Plug do zero, precisamos definir o módulo da aplicação.
-Atualize `lib/example.ex` para iniciar e superviosionar o Cowboy.
+Nós precisamos dizer para a nossa aplicação iniciar e supervisionar o servidor web Cowboy when o app estiver de pé.
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: Missing translation of `when`.

File: pt/lessons/specifics/plug.md [hunk 4]
```diff
@@ -72,40 +78,49 @@ Ela recebe um `%Plug.Conn{}` struct como seu primeiro argumento, e é esperado q
...
+
+Nós vamos fazer isso com a função [`Plug.Cowboy.child_spec/1`](https://hexdocs.pm/plug_cowboy/Plug.Cowboy.html#child_spec/1).
+
+Essa função espera três opções:
+
+* `:scheme` - HTTP ou HTTPS como um atom (`:http`, `:https`)
+* `:plug` - O módulo plug que deve ser usado como a interface para o servidor web. Você pode especificar o nome do módulo, como `MyPlug`, ou uma tupla com o nome do módulo e opções `{MyPlug, plug_opts}`, onde `plug_opts` é passada para a função `init/1` do seu módulos plug.
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: At the of the phrase, I believe there is a small problem with the plural of "módulos".

File: pt/lessons/specifics/plug.md [hunk 5]
```diff
@@ -145,31 +160,34 @@ Este é um mini Router, mas o código deve ser bastante auto-explicativo.
...
-Agora no navegador, digite `127.0.0.1:8080`.
+Agora no navegador, digite <http://127.0.0.1:8080>.
 Você deve ver `Welcome`.
-Então, digite `127.0.0.1:8080/waldo`, ou qualquer outro path. Isto deve retornar `Oops!` com uma resposta 404.
+Então, digite <http://127.0.0.1:8080/waldo>, ou qualquer outro path. Isto deve retornar `Oops!` com uma resposta 404.
 
 ## Adicionando outro Plug
 
-É comum criar Plugs para interceptar todas as requisições ou um subconjunto delas, para manipular lógicas comuns às requisições.
+É comum usar mais de um plug em uma única aplicação web, cada uma tendo sua própria responsabilidade. Por exemplo, nós podemos ter um plig que lida com roteamento, um plug que valida as requisições recebidas, um plug que autentica as requisições, etc. Nessa seção, nós vamos definir um plug para verificar os parâmetros das requisições recebidas e nós vamos ensinar a nossa aplicação a usar _ambos_ os plugs - o router e o plug de validação.
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: Typo in one of the plug words: "plig".

File: pt/lessons/specifics/plug.md [hunk 6]
```diff
@@ -237,60 +254,64 @@ defmodule Example.Router do
   plug(:dispatch)
 
   get("/", do: send_resp(conn, 200, "Welcome\n"))
-  post("/upload", do: send_resp(conn, 201, "Uploaded\n"))
+  get("/upload", do: send_resp(conn, 201, "Uploaded\n"))
   match(_, do: send_resp(conn, 404, "Oops!\n"))
 end
 ```
 
+Com esse código, nós estamos dizendo nossa aplicação para enviar as requisições recebidas através do plug `VerifyRequest` antes de passar pelo código no router. Através da chamada de função:
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: I believe "... nós estamos dizendo **à** nossa aplicação..." is better..?

File: pt/lessons/specifics/plug.md [hunk 7]
```diff
@@ -237,60 +254,64 @@ defmodule Example.Router do
...
+plug(
+  VerifyRequest,
+  fields: ["content", "mimetype"],
+  paths: ["/upload"]
+)
+```
+Nós automaticamente invocamos `VerifyRequest.init(fields: ["content", "mimetype"],
+paths: ["/upload"])`. Isso por sua vez passa as opções recebidas para a função `VerifyRequest.call(conn, opts)`.
+
+Let's take a look at this plug in action! Go ahead and crash your local server (rember, that's done by pressing `ctrl + c` twice). Then restart the server (`mix run --no-halt`).
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: Missing translation of the following paragraphs.

File: pt/lessons/specifics/plug.md [hunk 8]
```diff
@@ -351,6 +371,90 @@ Execute com o comando:
...
+    IO.puts "Reason:"
+    IO.inspect reason
+    IO.puts "Stack"
+    IO.inspect stack
+    send_resp(conn, conn.status, "Something went wrong")
+  end
+end
+```
+
+Nós vamos notar que no topo nós estamos adicionando `use Plug.ErrorHandler`. Esse plug agora captura qualquer error e então busca por uma função `handle_errors/2` para chamar. `handle_errors` apenas precisa aceitar uma `conn` como seu primeiro argumento e então um mapa com três itens (`:kind`, `:reason`, e `:stack`) como segundo argumento.
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: "... qualquer error..." => "... qualquer erro..."

**Comment by thiamsantos** [comment]
@anderkonzen could you please take another look?

File: pt/lessons/specifics/plug.md [hunk 9]
```diff
@@ -271,9 +271,9 @@ plug(
...
-Now go to <http://127.0.0.1:8080/upload> in your browser and you'll see that the page simply isn't working. We're not even getting our 'Oops!' message. Now let's add our required params by going to <http://127.0.0.1:8080/upload?content=thing1&mimetype=thing2>. Now we should see our 'Uploaded' message.
-It's not great that when we throw an error we don't get _any_ page, but we'll deal with how to handle errors with plugs later.
+Vamos ver como esse plug funciona em ação! Vá em frente e quebre seu servidor local (lembre-se, isso pode ser feito pressionando `ctrl + c` duas vezes). Então reinicie o servidor com (`mix run --no-halt`).
+Agora acesse <http://127.0.0.1:8080/upload> no seu navegador e você vai ver como a página simplesmente não está funcionando. Nós não estamos recebendo nossa mensagem 'Oops!'. Agora vamos adicionar os parâmetros obrigatórios por acessar <http://127.0.0.1:8080/upload?content=thing1&mimetype=thing2>. Agora nós devemos ver nossa mensagem 'Uploaded'. 
+Não é legal não receber _nenhuma_ página caso um erro occura, mas nós vamos como lidar com erros em plugs depois.
```
Comment (anderkonzen) [inline pt/lessons/specifics/plug.md]: There is a typo here: "... caso um erro **occura**"=> "ocorra".

Also, I think the end of the phrase could be translated to: "mas nós vamos lidar com como tratar erros com plug depois.".

Comment (thiamsantos) [reply pt/lessons/specifics/plug.md]: @anderkonzen thanks for the review! Sorry for the delay, I just pushed the suggested changes. Could you please take another look?

Comment (anderkonzen) [reply pt/lessons/specifics/plug.md]: There is still one tiny typo: `ocurra` => `ocorra`

The rest is OK!

Comment (thiamsantos) [reply pt/lessons/specifics/plug.md]: @anderkonzen :man_facepalming: sorry :joy: :joy:. I will fix it right now! Thanks!

Comment (thiamsantos) [reply pt/lessons/specifics/plug.md]: @anderkonzen done.

**Comment by anderkonzen** [comment]
Grrrreeaaat! Thanks @thiamsantos !

**Pull request merged**